classes.py

```diff
@@ -0,0 +1,372 @@
+        for a in self._actors:
+            actors += f' {a}'
+            if a != self._actors[-1]:
+                actors += f','
+        description = f'\nDesccription: {self._description}\n'
+        rating = f'Rating: {self._user_rating}'
+        return title+year+genre+director+actors+description+rating
+
+    
+
+class User:
+    """
+    Class User. Contains Attributes:
+
+    :param username: username
+    :type username: str
+
+    :param password: password
+    :type password: str
+
+    :param my_ratings: user's ratings, defaults to None
+    :type my_ratings: dict, key - title of the film, value - int rating
+    """
+    def __init__(self, id, username, password, my_ratings=None):
+        if id < 0 or type(id) != int:
+            raise NonPositiveIntValueError()
+        if username == '':
+            raise EmptyStringError()
+        if password == '':
+            raise EmptyStringError()
+        self._username = username
+        self._password = password
+        self._id = id
+        if my_ratings:
+            self._my_ratings = my_ratings
+        else:
+            self._my_ratings = {}
+
+    def id(self):
+        return self._id
+    def username(self):
+        return self._username
+    def password(self):
+        return self._password
+    def my_ratings(self):
+        return self._my_ratings
+    
+    def add_rating(self, film, rating):
+        """
+        Takes 3 parameters. Self, string(title of the film to be rated), int(rating of the film).
+        Checks if this film was already rated. Updates user's ratings. Calls a method for updating a film's rating.
+        """
+        old_rating = 0
+        if film.title() in self._my_ratings:
+            old_rating = self._my_ratings[film.title()]
+        self._my_ratings[film.title()] = rating
+        film.update_rating(rating, old_rating)
+        
+    
+class Portal():
+    """
+    Class Portal. Contains attributes:
+
+    :param films: films
+    :type films: list
+
+    :param users: users
+    :type users: list
+    """
+    def __init__(self,current_user, films, users):
+        if not current_user:
+            raise EmptyValueError()
+        if not films:
+            raise EmptyValueError()
+        if not users:
+            raise EmptyValueError()
+        self._current_user = current_user
+        self._films = films
+        self._users = users
+
+    def current_user(self):
+        return self._current_user
+    def films(self):
+        return self._films
+    def users(self):
+        return self._users
+    
+    def get_film_by_title(self, title):
+        """
+        Takes 2 parameters. Self, string(title of the film).
+        returns list of objects Film with matching titles.
+        """
+        films = self.films()
+        output = []
+        for film in films:
+            current_title = film.title()
+            if title.lower() in current_title.lower():
+                output.append(film)

```

Comment:
This is a brute-force search. Takes O(n). Will be inefficient for big collections. Consider using `dict` to get O(1) lookup complexity.